cardano-cli/src/Cardano/CLI/Shelley/Run/Read.hs

```diff
@@ -458,16 +467,56 @@ readFileTx fp = do
 -- while UnwitnessedCliFormattedTxBody is CLI formatted TxBody and
 -- needs to be key witnessed.
 
-data IncompleteTx
-  = UnwitnessedCliFormattedTxBody (InAnyCardanoEra TxBody)
-  | IncompleteCddlFormattedTx (InAnyCardanoEra Tx)
-
-readFileTxBody :: FilePath -> IO (Either CddlError IncompleteTx)
-readFileTxBody fp = do
-  eTxBody <- readFileInAnyCardanoEra AsTxBody fp
-  case eTxBody of
-    Left e -> fmap (IncompleteCddlFormattedTx . unCddlTx) <$> acceptTxCDDLSerialisation e
-    Right txBody -> return $ Right $ UnwitnessedCliFormattedTxBody txBody
+data IncompleteTx era
```

Comment:
What is the benefit of adding the `era` parameter? This has further complicated the code without any obvious benefit.